### PR TITLE
SCRUM-79-Verification-Screen

### PR DESCRIPTION
**Pull request by AlexeyBezrebriy**
## Describe your changes

- I added routes protection.
- I added verification screen and connect it to the backend.
image

## Issue ticket code (and/or) and link

- [SCRUM-79](https://codecraftersintership.atlassian.net/browse/SCRUM-79)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [ ] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/pages/NotFound/styles.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  backgroundColor: (theme) => theme.palette.background.default,
+  height: '100vh',
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'center',
+  alignItems: 'center',
+};
+
+export const headingStyles: SxProps<Theme> = {
+  fontSize: '36px',
```
Comment (asdelay) [inline src/pages/NotFound/styles.ts]: use font size from constant

File: src/pages/NotFound/styles.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  fontWeight: (theme) => theme.typography.fontWeightLight,
+  color: (theme) => theme.palette.text.primary,
+  marginBottom: 2,
+};
+
+export const messageStyles: SxProps<Theme> = {
+  fontSize: '18px',
```
Comment (asdelay) [inline src/pages/NotFound/styles.ts]: use font size from constants

File: src/pages/SignIn/useSignIn.ts [hunk 3]
```diff
@@ -30,19 +34,18 @@ const useSignIn = (): UseSignInReturnType => {
...
-      }
-
-      if (attemptCount + 1 >= maxAttempts) {
-        setIsLocked(true);
+        if (attemptCount + 1 >= maxTries) {
+          setIsLocked(true);
+        }
       }
     } catch (error) {
       throw new Error('Error occurred while sending verification code');
```
Comment (asdelay) [inline src/pages/SignIn/useSignIn.ts:51]: use i18n

File: src/pages/VerificationPage/useVerification.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    try {
+      setIsLocked(true);
+      const isSent = await sendVerificationCode(email)();
+
+      if (isSent) {
+        setTimeLeft(otpExpiryTime);
+      }
+    } catch (error) {
+      throw new Error('Error occurred while sending verification code');
```
Comment (asdelay) [inline src/pages/VerificationPage/useVerification.ts:76]: i18n

**Review by asdelay — COMMENTED** [review]
Fix font sizes I'll approve.

**Review by asdelay — APPROVED** [review]
Good job

File: src/pages/VerificationPage/useVerification.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  const [focusedIndex, setFocusedIndex] = useState<number | null>(null);
+  const [attemptCount, setAttemptCount] = useState(0);
+  const [isLocked, setIsLocked] = useState(false);
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation();
+
+  const email: string = location.state?.email || '';
+
+  const tick = 1000;
+  const maxTries = 5;
```
Comment (lyudad) [inline src/pages/VerificationPage/useVerification.ts:38]: move above useEffect

Comment (AlexeyBezrebriy) [reply src/pages/VerificationPage/useVerification.ts:38]: I can't move above useEffect because variables that i use inside the useEffect must be above

**Pull request merged**